tox_poetry_installer/utilities.py

```diff
@@ -243,7 +243,7 @@ def find_additional_deps(
             dep_name.lower(), packages, venv, allow_missing=[poetry.package.name]
         )
 
-    return dependencies
+    return dedupe_packages(dependencies)
```

Comment:
Rather than calling dedupe in a bunch of places throughout the codebase, could we instead call it just once on the [`packages`](https://github.com/enpaul/tox-poetry-installer/blob/devel/tox_poetry_installer/installer.py#L25) parameter to the `installer.install()` function? I'll admit I haven't done a full eval of all the secondary effects that may have, but that seems both cleaner and more efficient to me, so if it works I'd like to go with that approach instead.

Reply:
There is a bunch of logging/sanity checks along the way that make more sense if we're consistently doing that dedupe: e.g. `Identified {len(dev_deps)} development dependencies to install to env`